lib/OpenQA/Task/Asset/Limit.pm

```diff
@@ -89,8 +90,15 @@ sub _limit {
             my $age_in_seconds = ($now->epoch() - DateTime::Format::Pg->parse_datetime($asset->{t_created})->epoch());
             my $asset_name     = $asset->{name};
             if ($age_in_seconds >= $untracked_assets_storage_duration) {
+                my $storage_duration = $untracked_assets_storage_duration;
+                for my $pattern ($untracked_assets_patterns) {
```

Comment:
Shouldn't this be `keys %$untracked_assets_patterns`? But when I think about it, having a regex as HASH-key or INI-key (instead of a value) isn't that intuitive. Usually keys are just used as identifier and not to represent configuration values itself.

Reply:
I agree in general, but since we currently do plain substring matching  - I am fine with this simpler solution. Later we can add regexp as value if needed. (Or just replace regexp matching operator with substring search.)

Reply:
We discussed in Rocket Chat and since we're using INI there's actually not a good way to improve this anyways.